core/lib/Thelia/Action/Module.php

```diff
@@ -147,6 +153,61 @@ private function checkDeactivation($module)
         return true;
     }
 
+
+    /**
+     * Get dependencies of the current module and activate it if needed
+     *
+     * @param ModuleToggleActivationEvent $event
+     * @throws Exception
+     */
+    public function recursiveActivation(ModuleToggleActivationEvent $event)
+    {
+        if (null !== $module = ModuleQuery::create()->findPk($event->getModuleId())) {
+            try {
+                $moduleValidator = new ModuleValidator($module->getAbsoluteBaseDir());
+                $dependencies = $moduleValidator->getCurrentModuleDependencies();
+                foreach ($dependencies as $defMod) {
+                    $submodule = ModuleQuery::create()
+                        ->findOneByCode($defMod["code"]);
+                    if ($submodule && $submodule->getActivate() != BaseModule::IS_ACTIVATED) {
+                        $subevent = new ModuleToggleActivationEvent($submodule->getId());
+                        $subevent->setRecursive(true);
+                        $event->getDispatcher()->dispatch(TheliaEvents::MODULE_TOGGLE_ACTIVATION, $subevent);
+                    }
+                }
+            } catch (\Exception $ex) {
```

Comment:
If you check these methode : 
*  private function checkDeactivation($module)
*  private function checkActivation($module)

There is the same Exception handler so I assume is needed